File: src/java/jnpp/service/dto/clients/ProfessionalDTO.java
package jnpp.service.dto.clients;

import jnpp.service.dto.AddressDTO;
import jnpp.service.dto.IdentityDTO;

public class ProfessionalDTO extends ClientDTO {

    private String name;
    private IdentityDTO owner;

    public ProfessionalDTO(String login, String name, IdentityDTO owner, String email, AddressDTO address, String phone) {
        super(login, email, address, phone);
        this.name = name;
        this.owner = owner;
    }
        
    @Override
    public Type getType() {
        return ClientDTO.Type.PROFESIONAL;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public IdentityDTO getOwner() {
        return owner;
    }

    public void setOwner(IdentityDTO owner) {
        this.owner = owner;
    }
    
}

File: src/main/webapp/commons/dto/clients/professional-dto.js
class ProfessionalDTO extends ClientDTO {

    constructor(json) {
        super(json, ClientDTO.type.PROFESSIONAL);
        if (!json.owner || !json.name)
            throw new ParseException(json, typeof this);
        this.name = json.name;
        this.owner = new IdentityDTO(json.owner);
    }

}
